### PR TITLE
Self-written calculator and tests for it done

### PR DESCRIPTION
**Pull request by Bivis-s**
# Self-written calculator and tests for it done

**Task:**
Create a Calculator.java class containing 4 methods: sum, difference, division, multiplication
Use TestNG annotations and parameters, @BeforeMethod, @AfterMethod, dataProvider, run in parallel (via .xml file), RetryAnalyzer, TestListener, description, priority, invocationCount, threadPoolSize

## Calculator

My calculator has different modes:
image

Used for creating setting-classes:
image

Wich contains lambda-methods of operations:
image

The calculator is created using the factory-class:
image

Also, the calculator has two states: activated and deactivated, which is changed by appropriate methods. It means that calculator will throw an Exception every time when you try to do any operation when it is deactivated (_Calculator.java_):
image

How it works in console (_Run.java_): 
image

## Tests

There are 6 tests classes which tests every CalculatorSettings and CalculatorFactory
image
image

Some suites contains always-failing tests and that's ok:
image

All tests:
image

### CODE REVIEW

File: src/test/java/calculator_tests/broken_calculation_test/BrokenCalculationTest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    @AfterClass(alwaysRun = true)
+    public void discardCalculator() {
+        calculator = null;
+    }
+
+    @Test(groups = {"calculator", "positive", "broken_type"},
+            dataProvider = "correct addition data", dataProviderClass = TestData.class,
+            retryAnalyzer = CalculatorDeactivatedRetryAnalyzer.class)
+    public void casualTypeAdditionCalculationTest(double p1, double p2, double expectedResult)
```
Comment (vchesnakoutest) [inline src/test/java/calculator_tests/broken_calculation_test/BrokenCalculationTest.java:30]: this test fails with data:
                {-4.5, 5.6, 1.1},

java.lang.AssertionError: expected [1.1] but found [1.0999999999999996]
Expected :1.1
Actual   :1.0999999999999996

image

**Review by vchesnakoutest — COMMENTED** [review]
wow :) you did this work with a high patience 👍 

please fix failed test

File: src/test/java/calculator_tests/broken_calculation_test/BrokenCalculationTest.java (hunk 1 above)
Comment (Bivis-s) [reply src/test/java/calculator_tests/broken_calculation_test/BrokenCalculationTest.java:30]: These tests fail due to the calculator CASUAL setting logic, if I'll fix it, I'll get calculator like ACCURATE. Therefore, I see no reason to fix this.

**Comment by Bivis-s** [comment]
Added Allure plugin support
image

Casual Calculator renamed to Float Inaccurate Calculator
image

**Pull request merged**